src/inputs/internal/SelectFieldInput.tsx

```diff
@@ -142,35 +125,36 @@ export function SelectFieldInput<O, V extends Value>(props: SelectFieldInputProp
             // reset the field to its previous value. However, because we use a the Multiple Selection State manager,
             // then our `state.selectedKey` isn't set. So we need to properly reset the state ourselves.
             if (e.key === "Escape") {
-              // Triggering `Escape` is basically like re-selecting currently selected option, so do that if there is one.
-              state.selectionManager.setSelectedKeys(
-                selectedOptions.length > 0 ? [valueToKey(getOptionValue(selectedOptions[0]))] : [],
-              );
+              state.close();
+              resetField();
               return;
             }
 
             inputProps.onKeyDown && inputProps.onKeyDown(e);
           },
-          onBlur: () => {
+          onBlur: (e: React.FocusEvent) => {
+            // Do not call onBlur if readOnly or interacting within the input wrapper (such as the menu trigger button).
+            if (
+              inputProps.readOnly ||
+              (props.inputWrapRef.current && props.inputWrapRef.current.contains(e.relatedTarget as HTMLElement))
+            ) {
+              return;
+            }
+
             // We purposefully override onBlur here instead of using mergeProps, b/c inputProps.onBlur
             // goes into useComboBox's onBlur, which calls setFocused(false), which in useComboBoxState
             // detects a) there is no props.selectedKey (b/c we don't pass it), and b) there is an
             // `inputValue`, so it thinks it needs to call `resetInputValue()`.
-            //
-            // I assume we don't pass `selectedKey` b/c we support multiple keys.
-            if (isReadOnly) {
-              return;
-            }
             setIsFocused(false);
             maybeCall(onBlur);
             state.close();
 
-            // Always call `setSelectedKeys` onBlur with its existing selected keys..
-            // This ensures the field's `input.value` resets to what it should be in case it doesn't currently match.
-            state.selectionManager.setSelectedKeys(state.selectionManager.selectedKeys);
+            // Always call `resetField` onBlur, this ensures the field's `input.value` resets
+            // to what it should be in case it doesn't currently match.
+            resetField();
```

Comment:
The external caller won't need to be aware of this. This explicitly to reset the field's internal state. For instance, if I am typing to filter the options and then I blur without changing an option, it will reset to its previous value.